### PR TITLE
Re-download logic for VD content

### PR DESCRIPTION
**Pull request by diegodfrf**
|Related issue|
|---|
|#19940|

## Description

- Compilation without warnings in every supported platform
  - [ ] Linux
- [ ] Source installation

- [ ] Added unit tests (for new features)

### CODE REVIEW

File: src/shared_modules/content_manager/src/actionOrchestrator.hpp [hunk 1]
```diff
@@ -80,6 +82,11 @@ class ActionOrchestrator final
                     std::stoi(m_spBaseContext->spRocksDB->getLastKeyValue().second.ToString());
             }
 
+            if (offset == 0)
```
Comment (GabrielEValenzuela) [inline src/shared_modules/content_manager/src/actionOrchestrator.hpp:85]: What happens when `offset` is greater than `0`?

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp [hunk 2]
```diff
@@ -184,7 +189,27 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
...
+                    m_translationsDatabase->deleteAll();
+                    m_cvesDatabase->deleteAll();
+
+                    for (auto& [name, database] : m_feedsDatabases)
+                    {
+                        database->deleteAll();
+                    }
+
+                    const int offset = 0;
+                    const std::string url = "localhost/ondemand/" + topic_name + "?offset=" + std::to_string(offset);
```
Comment (GabrielEValenzuela) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:204]: I think we can make these lines more simple, changing to

```cpp
const std::string url = "localhost/ondemand/" + topic_name + "?offset=0";
```

And avoid a call of `std::to_string()`. What do you think about it?

**Review by GabrielEValenzuela — CHANGES_REQUESTED** [review]
Great job! I made a code review and left you some comments. 

Meanwhile, I going to start with the tests 😃

**Comment by GabrielEValenzuela** [comment]
# Report review 🟡 

Test are 🟢 

<details><summary> Test output </summary>

```bash
    Start 1: vulnerability_scanner_benchmark_tests
1/5 Test #1: vulnerability_scanner_benchmark_tests ...   Passed   19.63 sec
    Start 2: policy_manager_unit_test
2/5 Test #2: policy_manager_unit_test ................   Passed    0.01 sec
    Start 3: vulnerability_scanner_unit_tests
3/5 Test #3: vulnerability_scanner_unit_tests ........   Passed    0.60 sec
    Start 4: vulnerability_scanner_component_tests
4/5 Test #4: vulnerability_scanner_component_tests ...   Passed    0.04 sec
    Start 5: database_feed_manager_unit_test
5/5 Test #5: database_feed_manager_unit_test .........   Passed    2.89 sec

100% tests passed, 0 tests failed out of 5

Total Test time (real) =  23.17 sec
```

</details>

Testtol 🔴 

I see an infinite loop when I execute the testtool with the default parameters

<details><summary> Console output </summary>

```bash
ActionOrchestrator - Starting process
API offset to be used: 0
Output folders created.
FactoryContentUpdater - Starting process
Creating 'api' downloader
Creating 'raw' content decompressor
Version updater not needed
Downloaded content cleaner created
FactoryContentUpdater - Finishing process
ActionOrchestrator - Finishing process
ActionOrchestrator - Running process
APIDownloader - Starting
APIDownloader - Finishing - Download done successfully
SkipStep - Executing
PubSubPublisher - Data published
SkipStep - Executing
All files in the folder have been deleted.
Couldn't connect to server: -1
Action: Initiating scheduling action for test
ActionOrchestrator - Running process
APIDownloader - Starting
APIDownloader - Finishing - Download done successfully
SkipStep - Executing
PubSubPublisher - Data published
SkipStep - Executing
All files in the folder have been deleted.
Action: Initiating scheduling action for test
ActionOrchestrator - Running process
APIDownloader - Starting
APIDownloader - Finishing - Download done successfully
SkipStep - Executing
PubSubPublisher - Data published
SkipStep - Executing
All files in the folder have been deleted.
Action: Initiating scheduling action for test
ActionOrchestrator - Running process
APIDownloader - Starting
APIDownloader - Finishing - Download done successfully
SkipStep - Executing
PubSubPublisher - Data published
SkipStep - Executing
All files in the folder have been deleted.
[...]
```

</details>

I have a few questions:

- Is this behavior expected? 
- I think the socket is never created in the testtool, should I execute or create it before running the testtool?

File: src/shared_modules/content_manager/src/actionOrchestrator.hpp (hunk 1 above)
Comment (diegodfrf) [reply src/shared_modules/content_manager/src/actionOrchestrator.hpp:85]: The check is done in `OnDemandManager.cpp`, nd in this case if the `offset` is greater than `0` just keep the last value, `std::stoi(m_spBaseContext->spRocksDB->getLastKeyValue().second.ToString());`

**Comment by diegodfrf** [comment]
> # Report review 🟡
> Test are 🟢
> 
> Test output
> Testtol 🔴
> 
> I see an infinite loop when I execute the testtool with the default parameters
> 
> Console output
> I have a few questions:
> 
> * Is this behavior expected?
> * I think the socket is never created in the testtool, should I execute or create it before running the testtool?

- Is this behavior expected?
The maximum time is 60 seconds, after that, the testtool ends.

- I think the socket is never created in the testtool, should I execute or create it before running the testtool?
Yes, you must create the socket.

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp (hunk 2 above)
Comment (diegodfrf) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp]: Done

File: src/shared_modules/content_manager/src/actionOrchestrator.hpp (hunk 1 above)
Comment (GabrielEValenzuela) [reply src/shared_modules/content_manager/src/actionOrchestrator.hpp:86]: Thanks!

**Review by GabrielEValenzuela — APPROVED** [review]
LGTM! The overall structure is clear, and the code is easy to read and understand.

### Readability:
The variable names are descriptive, and the comments provide helpful insights into the logic.

### Testing:
The test run until the `segfault` was correct, I see that this `segfault` is out of our scope and needs to be fixed when the library has done it, so it is left for another PR.

<details><summary>Test report</summary>

#### Test requisites
- Run content_provider testtool

```bash 
wazuh-modulesd:content-updater:actionOrchestrator.hpp:40 ActionOrchestrator : Creating content updater orchestration
wazuh-modulesd:content-updater:executionContext.hpp:113 createRocksDB : API offset to be used: 0
wazuh-modulesd:content-updater:executionContext.hpp:143 createOutputFolder : The previous output folder: /tmp/testProvider will be removed.
wazuh-modulesd:content-updater:executionContext.hpp:157 createOutputFolder : Output folders created.
wazuh-modulesd:content-updater:factoryContentUpdater.hpp:43 create : FactoryContentUpdater - Starting process
wazuh-modulesd:content-updater:factoryDownloader.hpp:45 create : Creating 'api' downloader
wazuh-modulesd:content-updater:factoryDecompressor.hpp:72 create : Creating 'raw' content decompressor
wazuh-modulesd:content-updater:factoryVersionUpdater.hpp:50 create : Version updater not needed
wazuh-modulesd:content-updater:factoryCleaner.hpp:42 create : Downloaded content cleaner created
wazuh-modulesd:content-updater:factoryContentUpdater.hpp:60 create : FactoryContentUpdater - Finishing process
wazuh-modulesd:content-updater:actionOrchestrator.hpp:54 ActionOrchestrator : Content updater orchestration created
wazuh-modulesd:content-updater:actionOrchestrator.hpp:76 run : Running content update. Topic: test
wazuh-modulesd:content-updater:APIDownloader.hpp:37 download : APIDownloader - Starting
wazuh-modulesd:content-updater:APIDownloader.hpp:50 download : APIDownloader - Finishing - Download done successfully
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:pubSubPublisher.hpp:44 publish : PubSubPublisher - Data published
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:cleanUpContent.hpp:54 cleanUp : All files in the folder have been deleted.
wazuh-modulesd:content-updater:action.hpp:152 runActionOnDemand : Ondemand request - starting action for test
wazuh-modulesd:content-updater:actionOrchestrator.hpp:76 run : Running content update. Topic: test
wazuh-modulesd:content-updater:APIDownloader.hpp:37 download : APIDownloader - Starting
wazuh-modulesd:content-updater:APIDownloader.hpp:50 download : APIDownloader - Finishing - Download done successfully
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:pubSubPublisher.hpp:44 publish : PubSubPublisher - Data published
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:cleanUpContent.hpp:54 cleanUp : All files in the folder have been deleted.
==== ON SUCESS =====
wazuh-modulesd:content-updater:action.hpp:83 operator() : Initiating scheduling action for test
wazuh-modulesd:content-updater:actionOrchestrator.hpp:76 run : Running content update. Topic: test
wazuh-modulesd:content-updater:APIDownloader.hpp:37 download : APIDownloader - Starting
wazuh-modulesd:content-updater:APIDownloader.hpp:50 download : APIDownloader - Finishing - Download done successfully
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:pubSubPublisher.hpp:44 publish : PubSubPublisher - Data published
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:cleanUpContent.hpp:54 cleanUp : All files in the folder have been deleted.
wazuh-modulesd:content-updater:action.hpp:83 operator() : Initiating scheduling action for test
wazuh-modulesd:content-updater:actionOrchestrator.hpp:76 run : Running content update. Topic: test
wazuh-modulesd:content-updater:APIDownloader.hpp:37 download : APIDownloader - Starting
wazuh-modulesd:content-updater:APIDownloader.hpp:50 download : APIDownloader - Finishing - Download done successfully
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:pubSubPublisher.hpp:44 publish : PubSubPublisher - Data published
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:cleanUpContent.hpp:54 cleanUp : All files in the folder have been deleted.
wazuh-modulesd:content-updater:action.hpp:83 operator() : Initiating scheduling action for test
wazuh-modulesd:content-updater:actionOrchestrator.hpp:76 run : Running content update. Topic: test
wazuh-modulesd:content-updater:APIDownloader.hpp:37 download : APIDownloader - Starting
wazuh-modulesd:content-updater:APIDownloader.hpp:50 download : APIDownloader - Finishing - Download done successfully
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:pubSubPublisher.hpp:44 publish : PubSubPublisher - Data published
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:cleanUpContent.hpp:54 cleanUp : All files in the folder have been deleted.
wazuh-modulesd:content-updater:action.hpp:83 operator() : Initiating scheduling action for test
wazuh-modulesd:content-updater:actionOrchestrator.hpp:76 run : Running content update. Topic: test
wazuh-modulesd:content-updater:APIDownloader.hpp:37 download : APIDownloader - Starting
wazuh-modulesd:content-updater:APIDownloader.hpp:50 download : APIDownloader - Finishing - Download done successfully
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:pubSubPublisher.hpp:44 publish : PubSubPublisher - Data published
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:cleanUpContent.hpp:54 cleanUp : All files in the folder have been deleted.
wazuh-modulesd:content-updater:action.hpp:83 operator() : Initiating scheduling action for test
wazuh-modulesd:content-updater:actionOrchestrator.hpp:76 run : Running content update. Topic: test
wazuh-modulesd:content-updater:APIDownloader.hpp:37 download : APIDownloader - Starting
wazuh-modulesd:content-updater:APIDownloader.hpp:50 download : APIDownloader - Finishing - Download done successfully
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:pubSubPublisher.hpp:44 publish : PubSubPublisher - Data published
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:cleanUpContent.hpp:54 cleanUp : All files in the folder have been deleted.
wazuh-modulesd:content-updater:action.hpp:83 operator() : Initiating scheduling action for test
wazuh-modulesd:content-updater:actionOrchestrator.hpp:76 run : Running content update. Topic: test
wazuh-modulesd:content-updater:APIDownloader.hpp:37 download : APIDownloader - Starting
wazuh-modulesd:content-updater:APIDownloader.hpp:50 download : APIDownloader - Finishing - Download done successfully
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:pubSubPublisher.hpp:44 publish : PubSubPublisher - Data published
wazuh-modulesd:content-updater:skipStep.hpp:49 handleRequest : SkipStep - Executing
wazuh-modulesd:content-updater:cleanUpContent.hpp:54 cleanUp : All files in the folder have been deleted.
wazuh-modulesd:content-updater:onDemandManager.cpp:94 stopServer : Server stopped
AddressSanitizer:DEADLYSIGNAL
=================================================================
==47071==ERROR: AddressSanitizer: BUS on unknown address (pc 0x7fcd02c3173e bp 0x1007ffd191820c0 sp 0x7ffd191820b0 T0)
==47071==The signal is caused by a READ memory access.
==47071==Hint: this fault was caused by a dereference of a high value address (see register values below).  Disassemble the provided pc to learn which register was used.
AddressSanitizer:DEADLYSIGNAL
AddressSanitizer: nested bug in the same thread, aborting
```
</details>

Great job Diego!

**Pull request merged**